platform-sdk/swirlds-state-api/src/main/java/com/swirlds/state/spi/WritableSingletonStateBase.java

```diff
@@ -19,51 +19,43 @@
 import static java.util.Objects.requireNonNull;
 
 import edu.umd.cs.findbugs.annotations.NonNull;
+
 import java.util.ArrayList;
 import java.util.List;
-import java.util.Objects;
-import java.util.function.Consumer;
-import java.util.function.Supplier;
 
 /**
  * A convenient base class for mutable singletons.
  *
  * @param <T> The type
  */
-public class WritableSingletonStateBase<T> extends ReadableSingletonStateBase<T> implements WritableSingletonState<T> {
+public abstract class WritableSingletonStateBase<T> extends ReadableSingletonStateBase<T> implements WritableSingletonState<T> {
+
     /**
      * A sentinel value to represent null in the backing store.
      */
     private static final Object NULL_VALUE = new Object();
 
-    private final Consumer<T> backingStoreMutator;
+    /** Modified value buffered in this mutable state */
     private Object value;
-    /**
-     * Listeners to be notified when the singleton changes.
-     */
+
+    /** A list of listeners to be notified of changes to the state */
     private final List<SingletonChangeListener<T>> listeners = new ArrayList<>();
 
     /**
      * Creates a new instance.
      *
      * @param stateKey The state key for this instance
-     * @param backingStoreAccessor A {@link Supplier} that provides access to the value in the
-     *     backing store.
-     * @param backingStoreMutator A {@link Consumer} for mutating the value in the backing store.
      */
-    public WritableSingletonStateBase(
-            @NonNull final String stateKey,
-            @NonNull final Supplier<T> backingStoreAccessor,
-            @NonNull final Consumer<T> backingStoreMutator) {
-        super(stateKey, backingStoreAccessor);
-        this.backingStoreMutator = Objects.requireNonNull(backingStoreMutator);
+    public WritableSingletonStateBase(@NonNull final String serviceName, @NonNull final String stateKey) {
```

Comment:
All the states now accept "service name" and "state key" in constructors. These two values are used to compute the right VirtualMap key prefix for the state. It's done by finding the right state ID first using `StateUtils.stateIdFor()`, and then this ID is converted to bytes. Wouldn't it be easier to compute the ID once and store it in states rather than service name?

Reply:
Do you mean to change `WritableSingletonStateBase` constructor signature to `public WritableSingletonStateBase(@NonNull final Bytes stateId, @NonNull final String stateKey)`?